### DISCUSSION TITLE
Need help redefining rules to avoid left recursion and have precedence

### DISCUSSION THREAD

**Discussion by SlashScreen** [pest-parser/pest#816] [Q&A] [answered]
I have these rules here, to allow nested binary operations. I'm using 1==2 && 3==4 as a test input. 
However, if I write them like this, it becomes a left-recursion problem.
```
Expr = { UnaryExpr | BinaryExpr | FnCall | If | For | While | Term }
Term = {Int | "(" ~ Expr ~ ")" | Variable }

UnaryExpr = { UnOp ~ Term }
BinaryExpr = { BinaryExpr  ~ (BinOp ~ Expr)+ }
```
But if I write it like this, it has a precedence issue:
```
Expr = { UnaryExpr | BinaryExpr | FnCall | If | For | While | Term }
Term = {Int | "(" ~ Expr ~ ")" | Variable }

UnaryExpr = { UnOp ~ Term }
BinaryExpr = { Term  ~ (BinOp ~ Expr)+ }
```
As i, instead of parsing it as (1==2)&&(3==4)), it reads it as (1 == (2 && (3 == 4)). How do I rewrite this to read it as expected?
Adding parenthesis does work, but I want it to work without them.

**Comment by tomtau** [comment, marked as answer]
It can be done outside the grammar using `PrattParser`: https://pest.rs/book/precedence.html
